mlflow/store/artifact/artifact_repo.py

```diff
@@ -125,156 +142,68 @@ def download_artifacts(self, artifact_path, dst_path=None):
-
-        def async_download_artifact_dir(src_artifact_dir_path, dst_local_dir_path):
-            """
-            Initiate an asynchronous download of the artifact directory specified by
-            `src_artifact_dir_path` to the local filesystem directory specified by
-            `dst_local_dir_path`.
-
-            This implementation is adapted from
-            https://github.com/mlflow/mlflow/blob/a776b54fa8e1beeca6a984864c6375e9ed38f8c0/mlflow/
-            store/artifact/artifact_repo.py#L93.
-
-            :param src_artifact_dir_path: A relative, POSIX-style path referring to a directory of
-                                          of artifacts stored within the repository's artifact root
-                                          location. `src_artifact_dir_path` should be specified
-                                          relative to the repository's artifact root location.
-            :param dst_local_dir_path: Absolute path of the local filesystem destination directory
-                                       to which to download the specified artifact directory. The
-                                       downloaded artifacts may be written to a subdirectory of
-                                       `dst_local_dir_path` if `src_artifact_dir_path` contains
-                                       subdirectories.
-            :return: A tuple whose first element is the destination directory of the downloaded
-                     artifacts on the local filesystem and whose second element is a list of
-                     `InflightDownload` objects, each of which represents an inflight asynchronous
-                     download operation for a file in the specified artifact directory.
-            """
-            local_dir = os.path.join(dst_local_dir_path, src_artifact_dir_path)
-            inflight_downloads = []
-            dir_content = [  # prevent infinite loop, sometimes the dir is recursively included
-                file_info
-                for file_info in self.list_artifacts(src_artifact_dir_path)
-                if file_info.path != "." and file_info.path != src_artifact_dir_path
-            ]
-            if not dir_content:  # empty dir
-                if not os.path.exists(local_dir):
-                    os.makedirs(local_dir, exist_ok=True)
-            else:
-                for file_info in dir_content:
-                    if file_info.is_dir:
-                        inflight_downloads += async_download_artifact_dir(
-                            src_artifact_dir_path=file_info.path,
-                            dst_local_dir_path=dst_local_dir_path,
-                        )[1]
-                    else:
-                        inflight_downloads += async_download_artifact(
-                            src_artifact_path=file_info.path,
-                            dst_local_dir_path=dst_local_dir_path,
-                        )
-            return local_dir, inflight_downloads
```

Comment:
To me, this function is confusing because it does both download and listing artifacts.